service/history/shard/controller_test.go

```diff
@@ -728,6 +738,133 @@ func (s *controllerSuite) Test_GetOrCreateShard_InvalidShardID() {
 	s.ErrorIs(err, invalidShardIdUpperBound)
 }
 
+func (s *controllerSuite) TestShardLingerTimeout() {
+	shardID := int32(1)
+	s.config.NumberOfShards = 1
+	s.config.ShardLingerEnabled = func() bool {
+		return true
+	}
```

Comment:
```suggestion
	s.config.ShardLingerEnabled = dynamicconfig.GetBoolPropertyFn(true)
```
no one likes the static dynamic config functions :(